pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -638,6 +638,14 @@ func (t *target) needsRemediation(timeoutForMachineToHaveNode time.Duration) (bo
 		return false, nextCheck, nil
 	}
 
+	if t.MHC.Spec.MaxAge != nil {
+		machineCreationTime := t.Machine.CreationTimestamp.Time
+		if machineCreationTime.Add(t.MHC.Spec.MaxAge.Duration).Before(now) {
+			klog.V(3).Infof("%s: unhealthy: machine reached maximal age %v", t.string(), t.MHC.Spec.MaxAge.Duration)
```

Comment:
Beyond just logging a message does the MHC generate k8s events or have prometheus metrics?

We have some similar-ish node killer automation running in our OCP v3.11 environment and we found it very useful to create k8s events and have prometheus metrics for monitoring/tracking old nodes that are being deleted in our production environment.